oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/DefaultOAuth2User.java

```diff
@@ -69,8 +70,10 @@ public DefaultOAuth2User(Collection<? extends GrantedAuthority> authorities, Map
 		Assert.notEmpty(attributes, "attributes cannot be empty");
 		Assert.hasText(nameAttributeKey, "nameAttributeKey cannot be empty");
 		if (!attributes.containsKey(nameAttributeKey)) {
-			throw new IllegalArgumentException("Missing attribute '" + nameAttributeKey + "' in attributes");
+			throw new IllegalArgumentException("Missing nameAttributeKey '" + nameAttributeKey + "' in attributes");
 		}
+		Assert.notNull(attributes.get(nameAttributeKey), "Name attribute '" + nameAttributeKey + "' cannot be null");
```

Comment:
I think `Map#get` returing `null` implies that `Map#containsKey` returns `false`.

Would you please then simply replace the `containsKey` check with the `nonNull` check?

Please change the message also to:

```
Assert.notNull(attributes.get(nameAttributeKey), "Attribute value for '" + nameAttributeKey + "' cannot be null");
```